src/lls_model.jl

```diff
@@ -152,25 +187,31 @@ function NLPModels.residual!(nls::LLSModel, x::AbstractVector, Fx::AbstractVecto
 end
 
 function NLPModels.jac_structure_residual!(
-  nls::LLSModel,
+  nls::LLSModel{T, S, M1, M2},
   rows::AbstractVector{<:Integer},
   cols::AbstractVector{<:Integer},
-)
+) where {T, S, M1 <: AbstractMatrix, M2 <: AbstractMatrix}
   @lencheck nls.nls_meta.nnzj rows
   @lencheck nls.nls_meta.nnzj cols
   rows .= nls.A.rows
   cols .= nls.A.cols
   return rows, cols
 end
 
-function NLPModels.jac_coord_residual!(nls::LLSModel, x::AbstractVector, vals::AbstractVector)
+function NLPModels.jac_coord_residual!(nls::LLSModel{T, S, M1, M2}, x::AbstractVector, vals::AbstractVector) where {T, S, M1 <: AbstractMatrix, M2 <: AbstractMatrix}
   @lencheck nls.meta.nvar x
   @lencheck nls.nls_meta.nnzj vals
   increment!(nls, :neval_jac_residual)
   vals .= nls.A.vals
   return vals
 end
 
+function NLPModels.jac_residual(nls::LLSModel{T, S, M1, M2}, x::AbstractVector) where {T, S, M1 <: AbstractLinearOperator, M2 <: AbstractLinearOperator}
```

Comment:
why must M1 and M2 be operators here?

Reply:
The default `jac_residual` was being called here, creating a sparse matrix from the `rows`, `cols` and `vals` gotten from `jac_structure` and `jac_coord`. This doesn't work for linear operators.
If you think `jac_residual` should just return `nls.A` directly, we can open a new issue.

Reply:
Yes I see. I think you did it right but there are two things that are a bit weird:
1. the standard `jac_residual` returns a SparseMatrixCSC whereas `jac_structure_residual` and `jac_coord_residual` essentially give a COO format. This is a general comment for later. Perhaps all those functions should return a SparseMatrixCOO.
2. We also have a `jac_op_residual` which returns a linear operator. So I just wonder if it would have been more consistent to leave `jac_residual` unimplemented when `M1 <: AbstractLinearOperator` and only implement `jac_op_residual`. What do you think?

Reply:
2. Yes, indeed that makes more sense.